meson.build

```diff
@@ -0,0 +1,86 @@
+project(
+	'dunst',
+	'c',
+	version: '1.9.2',
+	license: 'MIT',
+	meson_version: '>=0.60.0',
+	default_options: [
+		'c_std=gnu99',
+		'warning_level=1',
+		'buildtype=release',
+	],
+)
+
+add_project_arguments([
+	'-pedantic',
+	'-Wno-gnu-zero-variadic-macro-arguments',
+	'-Wno-overlength-strings',
+], language: 'c')
+
+if get_option('buildtype').startswith('debug')
+	add_project_arguments('-DDEBUG_BUILD', language: 'c')
+endif
+
+cc = meson.get_compiler('c')
+
+cairo = dependency('cairo')
+glib = dependency('glib-2.0')
+gio = dependency('gio-2.0')
+gdk_pixbuf = dependency('gdk-pixbuf-2.0')
+pangocairo = dependency('pangocairo')
+x11 = dependency('x11')
+xinerama = dependency('xinerama')
+xext = dependency('xext')
+xrandr = dependency('xrandr', version: '>=1.5')
+xscrnsaver = dependency('xscrnsaver')
+systemd = dependency('systemd', required: get_option('systemd'))
+libnotify = dependency('libnotify', required: get_option('dunstify'))
+realtime = cc.find_library('rt')
+math = cc.find_library('m')
+wayland_client = dependency('wayland-client', required: get_option('wayland'))
+wayland_cursor = dependency('wayland-cursor', required: get_option('wayland'))
+
+c_version_arg = '-DVERSION="@0@"'.format(meson.project_version())
+
+subdir('src')
+
+install_data('dunstctl', install_dir: get_option('bindir'))
+install_data('dunstrc', install_dir: get_option('sysconfdir') / 'dunst')
+
+conf_data = configuration_data()
+conf_data.set('bindir', get_option('bindir'))
+conf_data.set('sysconfdir', get_option('sysconfdir'))
+
+configure_file(
+	input: 'org.knopwob.dunst.service.in',
+	output: '@BASENAME@',
+	configuration: conf_data,
+	install_dir: join_paths(get_option('datadir'), 'dbus-1/services'),
+)
+
+if systemd.found()
+	user_units_dir = systemd.get_pkgconfig_variable('systemduserunitdir')
```

Comment:
I have noticed a few notification or systemd service affiliated projects don't actually install the systemd service themselves if the dependency is found, they keep it optional:
https://gitlab.archlinux.org/archlinux/packaging/packages/seatd/-/blob/main/PKGBUILD?ref_type=heads#L50
https://gitlab.archlinux.org/archlinux/packaging/packages/mako/-/blob/main/PKGBUILD?ref_type=heads#L47
Should dunst have the same behavior?
seatd used to install the systemd service but has dropped installing it since.

Reply:
For now it's best to replicate the makefile behavior as much as possible. So take a look at the makefile and see what it does there